pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -389,18 +390,19 @@ func (l *LibvirtDomainManager) setMigrationResultHelper(failed bool, completed b
 	}
 
 	l.metadataCache.Migration.WithSafeBlock(func(migrationMetadata *api.MigrationMetadata, _ bool) {
-		if abortStatus != "" {
-			migrationMetadata.AbortStatus = string(abortStatus)
-		}
-
 		if failed {
 			migrationMetadata.Failed = true
 			migrationMetadata.FailureReason = reason
 		}
 		if completed {
 			migrationMetadata.Completed = true
-			now := metav1.Now()
-			migrationMetadata.EndTimestamp = &now
+		}
+
+		if abortStatus != "" {
+			migrationMetadata.AbortStatus = string(abortStatus)
+		} else {
+			// only mark the migration as complete if there was no abortion
```

Comment:
why is that? shouldn't we (as the commit title suggests) always set an end timestamp?

Reply:
Abortion is a special case, I think it makes a bit more sense to not set it because with this PR setting the EndTimestamp basically means that the migration completed while with abortion this is never true.

Also read this https://github.com/kubevirt/kubevirt/pull/13729#discussion_r1923599852.